core/src/index/CMakeLists.txt

```diff
@@ -50,6 +50,8 @@ include(ExternalProject)
 include(DefineOptionsCore)
 include(BuildUtilsCore)
 
+using_ccache_if_defined( KNOWHERE_USE_CCACHE )
```

Comment:
should inherit "MILVUS_USE_CCACHE" from top level:
if (MILVUS_USE_CCACHE)
    message(STATUS "knowhere use CCACHE")
    add_compile_definitions("MILVUS_USE_CCACHE")
endif ()